deepmd/pt/model/descriptor/repformer_layer.py

```diff
@@ -908,14 +961,24 @@ def _cal_hg(
         if not smooth:
             # nb x nloc
             # must use type_as here to convert bool to float, otherwise there will be numerical difference from numpy
-            invnnei = 1.0 / (epsilon + torch.sum(nlist_mask.type_as(g2), dim=-1))
+            if not use_sqrt_nnei:
+                invnnei = 1.0 / (epsilon + torch.sum(nlist_mask.type_as(g2), dim=-1))
+            else:
+                invnnei = 1.0 / (
+                    epsilon + torch.sqrt(torch.sum(nlist_mask.type_as(g2), dim=-1))
+                )
             # nb x nloc x 1 x 1
             invnnei = invnnei.unsqueeze(-1).unsqueeze(-1)
         else:
             g2 = _apply_switch(g2, sw)
-            invnnei = (1.0 / float(nnei)) * torch.ones(
-                (nb, nloc, 1, 1), dtype=g2.dtype, device=g2.device
-            )
+            if not use_sqrt_nnei:
+                invnnei = (1.0 / float(nnei)) * torch.ones(
+                    (nb, nloc, 1, 1), dtype=g2.dtype, device=g2.device
+                )
+            else:
+                invnnei = (1.0 / (float(nnei) ** 0.5)) * torch.ones(
```

Comment:
You can use [`torch.rsqrt`](https://pytorch.org/docs/stable/generated/torch.rsqrt.html).